src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+
+/**
+ * @struct ov_core_version_list_t
+ */
+typedef struct {
+    ov_core_version_t *versions;
+    size_t num_vers;
+}ov_core_version_list_t;
+
+/**
+ * @struct ov_core_t
+ */
+typedef struct ov_core ov_core_t;
+
+/**
+ * @struct ov_node_t
+ */
+typedef struct ov_node ov_node_t;
+
+/**
+ * @struct ov_output_node_t
+ */
+typedef struct ov_output_node ov_output_node_t;
+
+/**
+ * @struct ov_output_node_list_t
+ */
+typedef struct{
+    ov_output_node_t *output_nodes;
+    size_t num;
+} ov_output_node_list_t;
+
+/**
+ * @struct ov_model_t
+ */
+typedef struct ov_model ov_model_t;
+
+/**
+ * @struct ov_preprocess_t
+ */
+typedef struct ov_preprocess ov_preprocess_t;
+
+/**
+ * @struct ov_preprocess_input_info_t
+ */
+typedef struct ov_preprocess_input_info ov_preprocess_input_info_t;
+
+/**
+ * @struct ov_preprocess_input_tensor_info_t
+ */
+typedef struct ov_preprocess_input_tensor_info ov_preprocess_input_tensor_info_t;
+
+/**
+ * @struct ov_preprocess_output_info_t
+ */
+typedef struct ov_preprocess_output_info ov_preprocess_output_info_t;
+
+/**
+ * @struct ov_preprocess_output_tensor_info_t
+ */
+typedef struct ov_preprocess_output_tensor_info ov_preprocess_output_tensor_info_t;
+
+/**
+ * @struct ov_preprocess_input_model_info_t
+ */
+typedef struct ov_preprocess_input_model_info ov_preprocess_input_model_info_t;
+
+/**
+ * @struct ov_preprocess_input_process_steps_t
+ */
+typedef struct ov_preprocess_input_process_steps ov_preprocess_input_process_steps_t;
+
+typedef enum {
+    RESIZE_LINEAR,
+    RESIZE_CUBIC,
+    RESIZE_NEAREST
+} ov_preprocess_resize_algorithm_e;
+
+/**
+ * @struct ov_compiled_model_t
+ */
+typedef struct ov_compiled_model ov_compiled_model_t;
+
+/**
+ * @struct ov_infer_request_t
+ */
+typedef struct ov_infer_request ov_infer_request_t;
+
+/**
+ * @struct ov_tensor_t
+ */
+typedef struct ov_tensor ov_tensor_t;
+
+/**
+ * @struct ov_element_type_e
+ */
+typedef enum {
+    UNDEFINED = 0,  //!< Undefined element type
+    DYNAMIC,    //!< Dynamic element type
+    BOOLEAN,    //!< boolean element type
+    BF16,       //!< bf16 element type
+    F16,        //!< f16 element type
+    F32,        //!< f32 element type
+    F64,        //!< f64 element type
+    I4,         //!< i4 element type
+    I8,         //!< i8 element type
+    I16,        //!< i16 element type
+    I32,        //!< i32 element type
+    I64,        //!< i64 element type
+    U1,         //!< binary element type
+    U4,         //!< u4 element type
+    U8,         //!< u8 element type
+    U16,        //!< u16 element type
+    U32,        //!< u32 element type
+    U64         //!< u64 element type
+} ov_element_type_e;
+
+/**
+ * @struct ov_layout_t
+ */
+typedef char ov_layout_t[MAX_DIMENSION];
+
+/**
+ * @struct ov_shape_t
+ */
+typedef struct {
+    int ranks;
+    size_t dims[MAX_DIMENSION];
+} ov_shape_t;
+
+/**
+ * @struct ov_PartialShape_t
+ */
+typedef struct {
+    int ranks;
+    const char *dims[MAX_DIMENSION];
```

Comment:
why partial shape consists of chars?

Reply:
I suppose we need to introduce struct for dimension, and partial shape should be an array of dimensions.